src/main/io/rcdevice.c

```diff
@@ -256,27 +256,25 @@ void runcamDeviceSimulate5KeyOSDCableButtonRelease(runcamDevice_t *device, rcdev
     runcamDeviceSendRequestAndWaitingResp(device, RCDEVICE_PROTOCOL_COMMAND_5KEY_SIMULATION_RELEASE, NULL, 0, 200, 1, NULL, parseFunc);
 }
 
-static rcdeviceResponseParseContext_t* getWaitingResponse(timeUs_t currentTimeUs)
+static rcdeviceResponseParseContext_t* getWaitingResponse(timeMs_t currentTimeMs)
 {
     rcdeviceResponseParseContext_t *respCtx = rcdeviceRespCtxQueuePeekFront(&watingResponseQueue);
-    while (respCtx != NULL && respCtx->timeoutTimestamp != 0 && currentTimeUs > respCtx->timeoutTimestamp) {
-        if (respCtx->timeoutTimestamp != 0 && currentTimeUs > respCtx->timeoutTimestamp) {
-            if (respCtx->maxRetryTimes > 0) {
-                runcamDeviceSendPacket(respCtx->device, respCtx->command, respCtx->paramData, respCtx->paramDataLen);
-                respCtx->timeoutTimestamp = currentTimeUs + respCtx->timeout;
-                respCtx->maxRetryTimes -= 1;
-                respCtx = NULL;
-                break;
-            } else {
-                respCtx->result = RCDEVICE_RESP_TIMEOUT;
-                if (respCtx->parserFunc != NULL) {
-                    respCtx->parserFunc(respCtx);
-                }
-
-                // dequeue and get next waiting response context
-                rcdeviceRespCtxQueueShift(&watingResponseQueue);
-                respCtx = rcdeviceRespCtxQueuePeekFront(&watingResponseQueue);
+    while (respCtx != NULL && respCtx->timeoutTimestamp != 0 && currentTimeMs > respCtx->timeoutTimestamp) {
+        if (respCtx->maxRetryTimes > 0) {
+            runcamDeviceSendPacket(respCtx->device, respCtx->command, respCtx->paramData, respCtx->paramDataLen);
+            respCtx->timeoutTimestamp = currentTimeMs + respCtx->timeout;
```

Comment:
Sorry, I should have said this earlier, but you are assigning a `timeMs_t` to a `timeUs_ value.